clang/lib/AST/Decl.cpp

```diff
@@ -4695,14 +4699,14 @@ void FieldDecl::printName(raw_ostream &OS, const PrintingPolicy &Policy) const {
   DeclaratorDecl::printName(OS, Policy);
 }
 
-const FieldDecl *FieldDecl::findCountedByField() const {
-  const auto *CAT = getType()->getAs<CountAttributedType>();
+FieldDecl *FieldDecl::findCountedByField() const {
+  auto *CAT = getType()->getAs<CountAttributedType>();
   if (!CAT)
     return nullptr;
 
-  const auto *CountDRE = cast<DeclRefExpr>(CAT->getCountExpr());
-  const auto *CountDecl = CountDRE->getDecl();
-  if (const auto *IFD = dyn_cast<IndirectFieldDecl>(CountDecl))
+  auto *CountDRE = cast<DeclRefExpr>(CAT->getCountExpr());
+  auto *CountDecl = CountDRE->getDecl();
+  if (auto *IFD = dyn_cast<IndirectFieldDecl>(CountDecl))
```

Comment:
All of these changes can be reverted, right?

Reply:
Well, no. If I'm going to have to have the member returning a `const FieldDecl *` call the non-const function, then these can't be const, because that would result in trying to return a `const FieldDecl *`. Unless you would like me to use a `const_cast`, which I cannot understand WHY that is the preferred way of doing this. Bjarne literally hates `const_cast` but had to add it to make things work. And I'm not a big fan of it either.

Reply:
What I'm saying is that I don't see a reason why `findCountedByField()` should be a const member function returning a non-const reference. The usual pattern in Clang is to have a non-const member function which returns a non-const object and to have the const member function overload return a const object by calling the non-const overload with a single const_cast. As in: https://github.com/llvm/llvm-project/blob/4ed7bcb4a6100b2634580d9ef39e2f4cb3470fe9/clang/include/clang/AST/DeclCXX.h#L529

Reply:
But your original comment was that I could add the 'consts' back into that method, which I can't because I need it to return a non-const `FieldDecl`. If I revert it, I'll have to use a `const_cast` at the end, which defeats the whole purpose of re-adding the `consts`. If all you want is for me to make this a non-const method then fine.

Reply:
Actually, ignore that. In the new PR it's now all const.